### ISSUE TITLE
[BUG/SUPPORT] Repeat field + Custom field

### ISSUE THREAD

**Issue by jogelin** [formly-js/ngx-formly#890]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report
[ ] feature request
[X] support request
```

**Current behavior**
See example: [https://stackblitz.com/edit/formly-person-search?file=app%2Fperson-search-card.component.ts](https://stackblitz.com/edit/formly-person-search?file=app%2Fperson-search-card.component.ts)

Form containing a `FormArray` and a simple custom field (not a `FormGroup` like your example). The structure of fields/component is:
```
person-search-repeat-field (custom formly repeat field)
  |- person-search-field (custom formly field)
      |-person-search-card (custom form control)
```

Issues/Questions: 

**- Why a `FormGroup` is created by default if the field is custom ?** 
In my example, the `this.formControl` of `person-search-field` is a `FormGroup` by default. So I cannot map it to my custom element `person-search-card`. I checked your code and it looks like it is because I didn't specify a key...I don't have a key because it's a simple value containing a person object...

**- Why my model is not synchronize ?** 
In the example, you will see that the model and the form value are not synchronize in this case.

**- In `person-search-field`, I receive a `FormArray` in `this.form` but in you type `Field.form` as `FormGroup` only. Isn't better to put `FormGroup | FormArray` ?**

**Comment by thorgod** [comment]
To answer one of your questions in your stackblitz

1.
// ??? WHY HERE THE FORM CONTROL IS A FORMGROUP HERE ???
Use.
this.form

field 
```
import { FormGroup, AbstractControl } from '@angular/forms';
import { FormlyTemplateOptions, FormlyFieldConfig, FormlyFormOptions } from '../components/formly.field.config';
export declare abstract class Field {
    form: FormGroup;
    field: FormlyFieldConfig;
    model: any;
    options: FormlyFormOptions;
    readonly key: string;
    readonly formControl: AbstractControl;
    readonly to: FormlyTemplateOptions;
    readonly showError: boolean;
    readonly id: string;
    readonly formState: any;
}
```

2.
This is never called because it has no calls to it. 
  // WHY IS IT NEVER CALLED ?
  static createControl

**Comment by jogelin** [comment]
@thorgod thx for your investigation =

1. I am talking about this.formControl (and not this.form) and my question is more: why it is an instance of FormGroup in my case and not an instance of FormControl ? I feel that if the custom field is unknown, by default an FormGroup is created.

2. Directly related to the first remark, because my custom field is unknow, I could like to help formly by using the method static createControl which is already used in other control by formly. I think it should be generalized. When I checked the build form service, I do not enter in any cases of the creation of the field so by default I have a formGroup...but I would like to have a formControl instead...is it possible ?

Thx in advance for your help ;)

**Comment by aitboudad** [comment]
fixed in `2.0.1`, do not forget to remove:
```ts
// UGLY CODE TO FORCE USING FORM CONTROL INSTEAD OF FORMGROUP
...
```

:smile:

**Comment by jogelin** [comment]
@aitboudad Fixed : [https://stackblitz.com/edit/formly-person-search?file=app%2Fperson-search-card.component.ts](https://stackblitz.com/edit/formly-person-search?file=app%2Fperson-search-card.component.ts)

Thx @aitboudad you rocks !
